### PR TITLE
feat: Create Product Details Content component

### PR DESCRIPTION
**Pull request by icazevedo**
## What's the purpose of this pull request?

This PR adds the `ProductDetailsContent` component to the starter. 

## How does it work?

The component is an accordion that contains multiple pieces of information regarding the current product.

## How to test it?

Go to a product page and find the section below the product image. You should be able to open and close the accordion items. Also, check the [JIRA Ticket](https://vtex-dev.atlassian.net/jira/software/projects/FS/boards/632?selectedIssue=FS-546) for reference (it contains a link to Figma)
## Checklist

<em>You may erase this after checking them all :wink:</em>

**PR Title and Commit Messages**
- [x] PR title and commit messages follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification
  - Available prefixes: `feat`, `fix`, `chore`, `docs`, `style`, `refactor`, `perf`, and `test`

**PR Description**
- [x] Added a label according to the PR goal - `Breaking change`, `Features`, `Bug fixes`, `Chore`, `Documentation`, `Style changes`, `Refactoring`, `Performance`, and `Test`
- [x] Added the component, hook, or path name in-between backticks (\`\`) - *if applicable, e.g., `ComponentName` component, `useWindowDimensions` hook*

**Dependencies**
- [x] Committed the `yarn.lock` and `bun.lockb` file when there were changes to the packages

**Documentation**
- [x] PR description
- [x] Added to/Updated the Storybook - *if applicable*
- [x] For documentation changes, ping `@carolinamenezes` or `@PedroAntunesCosta` to review and update

### CODE REVIEW

File: src/components/sections/ProductDetails/ProductDetails.tsx [hunk 1]
```diff
@@ -197,12 +197,7 @@ function ProductDetails({ product: staleProduct }: Props) {
           />
         </section>
 
-        <section data-fs-product-details-content>
-          <article data-fs-product-details-description>
-            <h2 className="text__title-subsection">Description</h2>
-            <p className="text__body">{description}</p>
-          </article>
-        </section>
+        <ProductDetailsContent />
```
Comment (renatamottam) [inline src/components/sections/ProductDetails/ProductDetails.tsx:200]: I think we need to make some changes on this component here to fit PDP's grid.
1. It should follow `ImageGallery` columns, left aligned to main image:

|With Selectors|One Image Only|
|-|-|
|image|image|

2. `Accordion` component seems to have an extra padding left and right on both `data-fs-accordion-item-button` and `data-fs-accordion-item-panel` that shouldn't exist.

**Review by renatamottam — COMMENTED** [review]
Nice work, @icazevedo! I left some improvement suggestions :)

File: src/components/ui/Accordion/accordion.module.scss [hunk 2]
```diff
@@ -14,9 +14,6 @@
   --fs-accordion-item-button-color                  : var(--fs-color-text);
   --fs-accordion-item-button-bkg-color              : transparent;
 
-  --fs-accordion-item-button-padding-right-notebook : var(--fs-spacing-4);
-  --fs-accordion-item-button-padding-left-notebook  : var(--fs-accordion-item-button-padding-right-notebook);
```
Comment (renatamottam) [inline src/components/ui/Accordion/accordion.module.scss:18]: Remember to remove these from Token list on Stories list too

File: src/components/search/Filter/facets.module.scss [hunk 3]
```diff
@@ -18,6 +18,8 @@
   --fs-facets-accordion-item-button-text-size            : var(--fs-text-size-lead);
   --fs-facets-accordion-item-button-text-weight          : var(--fs-text-weight-regular);
   --fs-facets-accordion-item-button-line-height          : 1.5;
+  --fs-accordion-item-button-padding-right-notebook      : var(--fs-spacing-4);
+  --fs-accordion-item-button-padding-left-notebook       : var(--fs-accordion-item-button-padding-right-notebook);
```
Comment (renatamottam) [inline src/components/search/Filter/facets.module.scss:22]: Please, update these tokens from `--fs-accordion` to `--fs-facets-accordion`.
Remember to add them to the Stories too!

**Review by renatamottam — APPROVED** [review]
LGTM! 👏

File: src/components/sections/ProducDetailsContent/ProductDetailsContent.tsx [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+          as="article"
+          index={0}
+          isExpanded={indices.has(0)}
+          buttonLabel={
+            <h2 className="text__title-subsection">{labels.description}</h2>
+          }
+          data-fs-product-details-description
+          prefixId="product-details-content"
+        >
+          <p className="text__body">
```
Comment (filipewl) [inline src/components/sections/ProducDetailsContent/ProductDetailsContent.tsx:118]: I take that when we get to integrate this component we'll need to make changes to support more than 4 labels and their dynamic content? 

I'm curious to know how these label/content pairs are handled in the product's catalog.
🧐

**Review by filipewl — APPROVED** [review]
Looks _slick_ to me! 👏 

Left a suggestion to fix an accessibility test.

**Review by lucasfp13 — APPROVED** [review]
LGTM, considering the a11y fix pointed by @filipewl

File: src/components/sections/ProducDetailsContent/ProductDetailsContent.tsx [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+  }
+
+  const labels = { ...DEFAULT_LABELS, ...propLabels }
+
+  return (
+    <Section
+      className={styles.fsProductDetailsContent}
+      data-fs-product-details-content
+    >
+      <Accordion expandedIndices={indices} onChange={onChange}>
```
[the remark this answers is not in the document]
Comment (icazevedo) [reply src/components/sections/ProducDetailsContent/ProductDetailsContent.tsx:107]: Is this documented in some way?

[the remark this answers is not in the document]
Comment (icazevedo) [reply src/components/sections/ProducDetailsContent/ProductDetailsContent.tsx:107]: Also, if that breaks stores, shouldn't we make it required?

File: src/components/sections/ProducDetailsContent/ProductDetailsContent.tsx (hunk 4 above)
Comment (icazevedo) [reply src/components/sections/ProducDetailsContent/ProductDetailsContent.tsx:118]: I had this discussion with @renatamottam and we came up to the following conclusion: we're not supporting more than 4 labels. The last one will be the more flexible one, and the rest will always display specific parts of the product. That's done so we can assure better accessibility and use at the table and list sections. Also, that would make the product details component way too flexible, so in this case, developers would be better of by using the Accordion instead.

**Pull request merged**